guide/guide-basics.rst

```diff
@@ -45,28 +44,66 @@ The key difference between classical and quantum mechanics lies in the use of op
 
 To begin, let us create a blank ``Qobj``:
 
-.. ipython::
+.. testcode:: [basics]
+
+    print(Qobj())
+
+.. testoutput:: [basics]
+  :options: +NORMALIZE_WHITESPACE
+
+    Quantum object: dims = [[1], [1]], shape = (1, 1), type = bra
+    Qobj data =
+    [[0.]]
 
-    In [1]: Qobj()
 
 where we see the blank ``Qobj`` object with dimensions, shape, and data. Here the data corresponds to a 1x1-dimensional matrix consisting of a single zero entry.
 
 .. Hint:: By convention, Class objects in Python such as ``Qobj()`` differ from functions in the use of a beginning capital letter.
 
 We can create a ``Qobj`` with a user defined data set by passing a list or array of data into the ``Qobj``:
 
-.. ipython::
+.. testcode:: [basics]
+
+    print(Qobj([[1],[2],[3],[4],[5]]))
+
+.. testoutput:: [basics]
+  :options: +NORMALIZE_WHITESPACE
+
+    Quantum object: dims = [[5], [1]], shape = (5, 1), type = ket
+    Qobj data =
+    [[1.]
+    [2.]
+    [3.]
+    [4.]
+    [5.]]
 
-    In [1]: Qobj([[1],[2],[3],[4],[5]])
+.. testcode:: [basics]
 
-    In [2]: x = np.array([[1, 2, 3, 4, 5]])
+    x = np.array([[1, 2, 3, 4, 5]])
+    print(Qobj(x))
 
-    In [3]: Qobj(x)
+.. testoutput:: [basics]
+  :options: +NORMALIZE_WHITESPACE
 
-    In [4]: r = np.random.rand(4, 4)
+    Quantum object: dims = [[1], [5]], shape = (1, 5), type = bra
+    Qobj data =
+    [[1. 2. 3. 4. 5.]]
 
-    In [5]: Qobj(r)
+.. testcode:: [basics]
+  :skipif: True
 
+    r = np.random.rand(4, 4)
+    print(Qobj(r))
+
+.. testoutput:: [basics]
+  :skipif: True
+
+    Quantum object: dims = [[4], [4]], shape = (4, 4), type = oper, isherm = False
+    Qobj data =
+    [[0.80891494 0.80278898 0.33983968 0.04215302]
+     [0.98991331 0.22610373 0.70347075 0.42928218]
+     [0.60515625 0.09007662 0.63876191 0.94787943]
+     [0.19734676 0.97222882 0.68423903 0.3205153 ]]
```

Comment:
> From the rendered version, it is a bit hard to distinguish which lines are inputs and which are outputs. Is there a way to make it clearer?
Maybe I can try comments but not inherently.

> Just to verify if I understand it correctly, because it is random, this won't need a test, so your skipped it.
Yeah, maybe I can try random seed.